src/functions/handler.rb

```diff
@@ -0,0 +1,97 @@
+require 'json'
+
+
+GENRE_FILE = 'genres.json'.freeze
+MUSIC_FILE = 'music_albums.json'.freeze
+MUSIC_GENRE_FILE = 'music_albums_genres.json'.freeze
+
+def write_json(array, file_path)
+  opts = {
+    array_nl: "\n",
+    object_nl: "\n",
+    indent: '  ',
+    space_before: ' ',
+    space: ' '
+  }
+  File.write(file_path, JSON.generate(array, opts))
+end
+
+def on_entry(app)
+  check_music_files
+  read_genres(app)
+  read_albums(app)
+  read_music_genre(app)
+end
+
+def on_exit(app)
+  store_genres(app)
+  store_albums(app)
+  save_music_genre(app)
+end
+
+# CHECK IF FILE EXIST
+def check_music_files
+  File.write(GENRE_FILE, '{}') unless File.exist?(GENRE_FILE)
+  File.write(MUSIC_FILE, '{}') unless File.exist?(MUSIC_FILE)
+  File.write(MUSIC_GENRE_FILE, '{}') unless File.exist?(MUSIC_GENRE_FILE)
+end
+
+def read_json(file_path)
+  return unless File.exist?(file_path)
+
+  file = File.read(file_path)
+  JSON.parse(file)
+end
+
+# Store Genres
+def store_genres(app)
+  array = []
+  app.genres.each do |genre|
+    array.push(name: genre.name)
+  end
+  write_json(array, GENRE_FILE)
+end
```

Comment:
[OPTIONAL] You can specify here the type of the file the genre is being assigned to ex. book, music, game

Reply:
Thank you @amexabee , I have made the changes as suggested.